lib/api/dsl/methods.js

```diff
@@ -109,6 +109,72 @@ var geoUtil = {
   },
 
   /**
+   * Construct a valid usable BBox
+   *
+   * @param {Object} the given object
+   * @return {Object} the valid usable BBox object
+   */
+  constructPolygon: function(geoFilter) {
+    var i,
+      tmp,
+      point,
+      entry,
+      polygon = [];
+
+    if (geoFilter.points === undefined) {
+      throw new BadRequestError('No point list found');
+    }
+
+    if (!_.isArray(geoFilter.points)) {
+      throw new BadRequestError('A polygon must be in array format');
+    }
+
+    if (geoFilter.points.length < 3) {
+      throw new BadRequestError('A polygon must have at least 3 points');
+    }
+
+    /*eslint guard-for-in: 0 */
+    for (i in geoFilter.points) {
+      entry = geoFilter.points[i];
+
+      if (_.isArray(entry)) {
+        if (entry.length >= 2) {
+          point = {lat: entry[0], lon: entry[1]};
+        } 
+        else {
+          throw new BadRequestError('Unable to parse some point coordinates into polygon');
```

Comment:
suggestion: if the parsed document is really complex or if there is lots of points, it may be useful for the user to return the incorrect point in the exception, to ease diagnostics.
Something along these lines: 'Unable to parse the following coordinate: ' + entry

What do you think?